compiler/rustc_interface/src/lib.rs

```diff
@@ -24,8 +24,6 @@ mod proc_macro_decls;
 mod queries;
 pub mod util;
 
-pub use callbacks::setup_callbacks;
```

Comment:
Looks like Geobacter uses this too: https://github.com/geobacter-rs/geobacter/blob/b569ef15dc7d49816e3ae5ea730d09fc94ee7e4b/runtime-core/src/codegen/worker/mod.rs#L86 It's been almost three years since they last updated their rustc fork though it seems.

Reply:
I will just remove the relevant commit, it's not that important.

Are there any basic principles I can follow to understand what APIs might be used outside of the repository? I've had trouble with things in `rustc_driver_impl` and `rustc_interface`.

Reply:
> Are there any basic principles I can follow to understand what APIs might be used outside of the repository?

I did a github search for the function name and `language:rust` and the manually ignored all search results for rustc forks.

Reply:
Right, so that's a 100% descriptive approach, but it seems like there's nothing prescriptive that can be relied on? "Just don't change anything that anyone is actually using, as determined by GitHub search" is not a good state of affairs. It's very easy to violate unintentionally, and gives no guidance on how to change things that are relied on.

Reply:
Usually we can remove everything that is not used in rustc and doesn't have a comment explaining which external tool needs it.

Reply:
My experience is that I do that, and then someone (often @bjorn3) complains, and so then I put it back and add that comment :/

Reply:
#116827 is the most recent example.